Jenkinsfile

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/groovy
+
+/*
+ * Copyright (C) 2016-2017 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+// Runs a shell command in the vagrant vm
+def runInVagrant = { String command ->
+    sh "cd ${env.WORKSPACE} && vagrant ssh -c 'cd dbus-proxy && ${command}'"
+}
+
+def shutdownVagrant = {
+    // In case the destroy step fails (happens sometimes with hung up ssh connections)
+    // we retry a couple of times to make sure it shuts down and is destroyed.
+    retry(5) {
+        sh "cd ${env.WORKSPACE} && vagrant destroy -f"
+    }
+}
+
+node {
+    try {
+
+        // Stages are subtasks that will be shown as subsections of the finiished build in Jenkins.
+        stage('Download') {
+            // Delete old files
+            sh 'rm -rf .[^.] .??* *'
+            // Checkout the git repository and refspec pointed to by jenkins
+            checkout scm
+            // Update the submodules in the repository.
+            sh 'git submodule update --init'
+        }
+
+        stage('StartVM') {
+            // Calculate available amount of RAM (excluding Swap)
+            String gigsramStr = sh (
+                script: 'free -g | grep "Mem: " | tail -n1 | awk \'{ print $2 }\'',
+                returnStdout: true
+            )
+            int gigsram = gigsramStr.trim() as Integer
+            // Only use half of the available memory if it is more than 2GB
+            if (gigsram >= 2) {
+                gigsram = gigsram / 2
+                println "Will set VAGRANT_RAM to ${gigsram}"
+            }
+
+            // Start the machine (destroy it if present) and provision it
+            shutdownVagrant()
+            withEnv(["VAGRANT_RAM=${gigsram}",
+                     "APT_CACHE_SERVER=10.8.36.16"]) {
+                sh "cd ${env.WORKSPACE} && vagrant up"
+            }
+        }
+
+        // Configure the software with cmake
+        stage('Configure') {
+            String buildParams = "-DENABLE_COVERAGE=ON "
+            buildParams       += "-DENABLE_USER_DOC=ON -DENABLE_API_DOC=ON "
+            buildParams       += "-DENABLE_SYSTEMD=ON -DENABLE_PROFILING=ON "
+            buildParams       += "-DENABLE_TEST=ON -DENABLE_EXAMPLES=ON"
```

Comment:
These parameters should be updated to reflect the ones available for dbus-proxy (these are copied from SoftwareContainer).